viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
-# ------------- WAVE 1 --------------------
+#
+#  ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
 
+#   Check to see if there is a title, genre, rating
+    if not title or not genre or not rating:
+        return None
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+#   Add movie to user's watched list
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+#   Add movie to user's watchlist list
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+#   Check to see if movie's in watchlist, and put into watched
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum = 0
+    counter = 0
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    for movies in user_data["watched"]:
+        sum += movies["rating"]
+        counter += 1
+    
+    return sum/counter
+    
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    largest_value = 0
+    most_common_genre = ""
+
+    if not user_data["watched"]:
+        return None
+    
+#   Count frequency of genre
+    for movies in user_data["watched"]:
+        if movies["genre"] not in genre_dict:
+            genre_dict[movies["genre"]] = 1
+        else:
+            genre_dict[movies["genre"]] += 1
+
+#   Find the most common genre
+    for genre, freq in genre_dict.items():
+        if freq > largest_value:
+            largest_value = freq
+            most_common_genre = genre
+    
+    return most_common_genre
+
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_movies = []
+
+#   Create a list of movies that friends have
+    for watched_list in user_data["friends"]:
+        for movies in watched_list["watched"]:
+            friend_movies.append(movies)
+
+#   Check to see if there are common movies between user movies and friends movies
+#   Adds uncommon movies for user
+    for user_movies in user_data["watched"]:
+        if user_movies not in friend_movies:
+            unique_movies.append(user_movies)
+
+    return unique_movies

```

Comment:
👍 Nice work! The rest of the functions use this `friend_movies` list elsewhere and would be great to add this as a helper function.